docs/source/features/parallelisms.rst

```diff
@@ -80,10 +72,36 @@ The distributed optimizer in NeMo is built on top of
 `DistributedFusedAdam <https://github.com/NVIDIA/apex/blob/master/apex/contrib/optimizers/distributed_fused_adam.py>`_
 from Apex.
 
+Fully-Shared Data Parallelism (FSDP)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+NeMo supports Fully-Sharded Data Parallelism (FSDP) that shards parameter gradients and low-precision parameters for computation on top of the model states that Distributed optimizer shards (optimizer states and high-precision parameters).
+Since FSDP shards the entire model states, it ensures linear model state memory saving with increasing DP size.
+FSDP can be preferred for the LLM training with unbalanced workload between pipeline stages (or Transformer layers) or with a large vocabulary size, where pipelining would cause huge computation bubbles due to the workload imbalance.
```

Comment:
FSDP is the preferred approach for training LLMs when there is an unbalanced workload between pipeline stages (or Transformer layers) or when dealing with a large vocabulary size. In such cases, using pipelining could lead to significant computation bubbles due to the workload imbalance.